docs/rackhd/install_os.rst

```diff
@@ -0,0 +1,128 @@
+dnsServers          Array            *optional*   If specified, this contains an array of string, each elements is the Domain Name Server, the first one will be primary, others are alternative.
+networkDevices      Array            *optional*   The static IP setting for network devices after OS installation. If it is omitted, null or empty, RackHD will not touch any network devices setting, so all network devices remain at the default state (usually default is DHCP).If there is multiple setting for same device, RackHD will choose the last one as the final setting, both ipv4 and ipv6 are supported here.
+rootSshKey          String           *optional*   The SSH key that will be appended to target OS.
+installDisk         String/Number    *optional*   *installDisk* is to specify the target disk which the OS will be installed on. It can be a string or a number. **For string**, it is a disk path that the OS can recongize, its format varies with OS. For example, "/dev/sda" or "/dev/disk/by-id/scsi-36001636121940cc01df404d80c1e761e" for CentOS/RHEL, "t10.ATA_____SATADOM2DSV_3SE__________________________20130522AA0990120088" or "naa.6001636101840bb01df404d80c2d76fe" or "mpx.vmhba1:C0:T1:L0" or "vml.0000000000766d686261313a313a30" for ESXi. **For number**, it is a RackHD generated disk identifier (it could be obtained from "driveId" catalog). If *installDisk* is omitted, RackHD will assign the default disk by order: SATADOM -> first disk in "driveId" catalog -> "sda" for Linux OS or "firstdisk" for ESXi.
+kvm                 Boolean          *optional*   The value is *true* or *false* to indicates to install kvm or not, default is **false**.
+switchDevices       Array            *optional*   **(ESXi only)** If specified, this contains an array of objects with switchName and uplinks (optional) parameters.  If *uplinks* is omitted, null or empty, the vswitch will be created with no uplinks.
+postInstallCommands Array            *optional*   **(ESXi only)** If specified, this contains an array of string commands that will be run at the end of the post installation step.  This can be used by the customer to tweak final system configuration.
+
+=================== ================ ============ ============================================
+
+For **users** in payload:
+
+=========== ======== ============ ============================================
+Parameters  Type     Flags        Description
+=========== ======== ============ ============================================
+name        String   **required** The name of user
+password    String   **required** The password of user, it could be clear text, RackHD will do encryption before store it into kickstart file.
```

Comment:
For ESXi 5.5
Your user password must meet the following length requirements.
■
Passwords containing characters from one or two character classes must be at least eight characters long.
■
Passwords containing characters from three character classes must be at least seven characters long.
■
Passwords containing characters from all four character classes must be at least six characters long.
https://pubs.vmware.com/vsphere-50/index.jsp#com.vmware.vsphere.security.doc_50/GUID-DC96FFDB-F5F2-43EC-8C73-05ACDAE6BE43.html?resultof=%2522%2570%2561%2573%2573%2577%256f%2572%2564%2522%2520


For ESXi6:
The password for administrator@vsphere.local must meet the following requirements:
■
At least 8 characters
■
At least one lowercase character
■
At least one numeric character
■
At least one special character
The password for administrator@vsphere.local cannot be more than 20 characters long. Only visible ASCII characters are allowed. That means, for example, that you cannot use the space character.

http://pubs.vmware.com/vsphere-60/index.jsp#com.vmware.vsphere.security.doc/GUID-4BDBF79A-6C16-43B0-B0B1-637BF5516112.html?resultof=%2522%2550%2561%2573%2573%2577%256f%2572%2564%2522%2520%2522%2570%2561%2573%2573%2577%256f%2572%2564%2522%2520%2522%2552%2565%2571%2575%2569%2572%2565%256d%2565%256e%2574%2573%2522%2520%2522%2572%2565%2571%2575%2569%2572%2522%2520